src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -241,6 +241,10 @@ public static class Avx
         /// </summary>
         public static byte Extract(Vector256<byte> value, byte index)
         {
+            if (!IsSupported)
```

Comment:
I don't believe we want any methods to work when `IsSupported` is false, even helpers.

We should also investigate if this is really better than (the below is what all three major native compilers do):
```asm
vpextrb dst, src, index       ; When index is into the lower 128-bits
```
or
```asm
vextractf128 dst, src, 0x1    ; When index is into the upper 128-bits
vpextrb dst, src, index
```

Reply:
> We should also investigate if this is really better than

Actually, RyuJIT also generates the above code as well as native compilers. The managed implementation here is just for the non-constant fallback. So, I believe this is really better than calling into one (lower 128-bits) or two (upper 128-bits) large jump-tables.

Reply:
Hmmm. I'm not so sure here. I always forget that the jit implementation exists (when I see the managed implementation here) and this is one of the smaller jump tables that we could generate (considering we only need to handle 32 cases max)

Reply:
I will say that, as a min bar, we should add a comment indicating that this is the software fallback.

However, I think that having this "different" software fallback goes against the principle of ensuring the helper indirect invocations execute the same instructions that the hardware accelerated path uses.

As it is now, we need to ensure that the software behavior is semantically equivalent to the hardware instructions that would have been executed (and it has already had to have been fixed a number of times due to minor issues).

Reply:
> However, I think that having this "different" software fallback goes against the principle of ensuring the helper indirect invocations execute the same instructions that the hardware accelerated path uses.

I agree; I think there's a lot of value in the uniformity of the current non-constant approach.

Reply:
https://github.com/dotnet/coreclr/issues/18743

Reply:
Hmm, some native compilers (i.e., clang) also optimize the non-const Extract/Insert to the memory operations. It would be better to get some data when we unify the implementation.

Reply:
@fiigii, I think the point is that the value of uniformity far outweighs any slight perf increase for the non-constant case (especially considering that the `non-constant` case already takes a perf hit from the call indirection/etc and is expected to be an edge case anyways).

For the constant case, all three compilers definitely emit the SIMD instructions.

Reply:
Right, I just meant the non-const situations.